File: src/main/kotlin/solutionsJava/WeirdAlgorithm.java
import java.util.Scanner;

public class WeirdAlgorithm {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        long n = scanner.nextInt();

        StringBuilder res = new StringBuilder(String.valueOf(n));

        while (n != 1) {
            if (n % 2 == 0) {
                n /= 2;
            } else {
                n = n * 3 + 1;
            }

            res.append(" ").append(n);
        }

        System.out.println(res);
    }
}

File: src/main/kotlin/solutions/WeirdAlgorithm.kt
package uz.safix.solutions

import java.util.Scanner

fun main() {
    val scanner = Scanner(System.`in`)
    var n = scanner.nextLong()

    val res = StringBuilder("$n")

    while (n != 1L) {
        if (n % 2 == 0L) {
            n /= 2
        } else {
            n = n * 3 + 1
        }

        res.append(" $n")
    }

    println(res)
}

